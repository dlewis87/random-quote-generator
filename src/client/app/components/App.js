import React from 'react'

const App = (props) => (
	<div style={props.backgroundColor}>
		<div className="container">
		  <div className="row">
		    <div className="col-xs-12">
		      <h1 className="text-center">Random Quote Generetor</h1>
		      <h2 className="text-center">Quotes are so inspiring...</h2>      
		    </div>    
		  </div>
		  <hr />
		  <div className="row">
		    <div className="col-xs-12">
		      <div className="text-center quote-box"  >
		        <div className="quote">
		        	<h3 className="text-left">
		        		<i className="fa fa-quote-left fa-2x"></i> 
		        	</h3>
		        	<p>{props.quote.quote}</p>
		        	<h3 className="text-right">
		        		<i className="fa fa-quote-right fa-2x"></i> 
		        	</h3>	        	
		        </div>
		        <p className="author">{props.quote.speaker}</p>
		      </div>
		      <hr />
		      <button id="getQuote" className="btn btn-default" onClick={props.submitNewQuote} >
		        Get Quote
		      </button>
		      <a 
		      	className="btn btn-primary" 
		      	id="twitter" 
		      	href={"https://twitter.com/intent/tweet?text=" + props.quote.quote} 
		      	target="_blank"
		      	> 
		      	<i className="fa fa-twitter"></i> Tweet
			  </a>
		    </div>
		  </div>
		</div>
	</div>
);

export default App;