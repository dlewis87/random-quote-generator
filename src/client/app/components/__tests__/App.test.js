import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
  const getNewQuote = jest.fn();

  const props = {
    quote: { quote: 'test', speaker: 'test' },
    backgroundColor: 'blue',
    getNewQuote
  };
  const app = shallow(<App {...props} />);

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });

  it('shows initial quote', () => {
    expect(app.find('.quote-content').text()).toContain('test');
  });

  it('shows initial speaker', () => {
    expect(app.find('.author').text()).toContain('test');
  });

  it('has background colour set from initial props', () => {
    expect(app.get(0).props.style).toHaveProperty('backgroundColor', 'blue');
  });

  describe('New quote button', () => {
    beforeEach(() => {
      app.find('#getQuote')
        .simulate('click');
    });

    it('calls new quote function', () => {
      expect(getNewQuote).toHaveBeenCalled();
    });
  });
});