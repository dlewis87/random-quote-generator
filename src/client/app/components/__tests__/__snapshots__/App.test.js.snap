// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    backgroundColor="blue"
    getNewQuote={[Function]}
    quote={
      Object {
        "quote": "test",
        "speaker": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-12"
          >
            <h1
              className="text-center"
            >
              Random Quote Generator
            </h1>
            <h2
              className="text-center"
            >
              Quotes are so inspiring...
            </h2>
          </div>
        </div>
        <hr />
        <div
          className="row quote-container"
        >
          <div
            className="col-xs-12"
          >
            <div
              className="text-center quote-box"
            >
              <div
                className="quote"
              >
                <h3
                  className="text-left"
                >
                  <i
                    className="fa fa-quote-left fa-2x"
                  />
                </h3>
                <p
                  className="quote-content"
                >
                  test
                </p>
                <h3
                  className="text-right"
                >
                  <i
                    className="fa fa-quote-right fa-2x"
                  />
                </h3>
              </div>
              <p
                className="author"
              >
                test
              </p>
            </div>
            <hr />
            <button
              className="btn btn-default"
              id="getQuote"
              onClick={[Function]}
            >
              Get Quote
            </button>
            <a
              className="btn btn-primary"
              href="https://twitter.com/intent/tweet?text=test"
              id="twitter"
              target="_blank"
            >
              <i
                className="fa fa-twitter"
              />
               Tweet
            </a>
          </div>
        </div>
      </div>,
      "style": Object {
        "backgroundColor": "blue",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <div
              className="col-xs-12"
            >
              <h1
                className="text-center"
              >
                Random Quote Generator
              </h1>
              <h2
                className="text-center"
              >
                Quotes are so inspiring...
              </h2>
            </div>
          </div>,
          <hr />,
          <div
            className="row quote-container"
          >
            <div
              className="col-xs-12"
            >
              <div
                className="text-center quote-box"
              >
                <div
                  className="quote"
                >
                  <h3
                    className="text-left"
                  >
                    <i
                      className="fa fa-quote-left fa-2x"
                    />
                  </h3>
                  <p
                    className="quote-content"
                  >
                    test
                  </p>
                  <h3
                    className="text-right"
                  >
                    <i
                      className="fa fa-quote-right fa-2x"
                    />
                  </h3>
                </div>
                <p
                  className="author"
                >
                  test
                </p>
              </div>
              <hr />
              <button
                className="btn btn-default"
                id="getQuote"
                onClick={[Function]}
              >
                Get Quote
              </button>
              <a
                className="btn btn-primary"
                href="https://twitter.com/intent/tweet?text=test"
                id="twitter"
                target="_blank"
              >
                <i
                  className="fa fa-twitter"
                />
                 Tweet
              </a>
            </div>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-xs-12"
            >
              <h1
                className="text-center"
              >
                Random Quote Generator
              </h1>
              <h2
                className="text-center"
              >
                Quotes are so inspiring...
              </h2>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="text-center"
                >
                  Random Quote Generator
                </h1>,
                <h2
                  className="text-center"
                >
                  Quotes are so inspiring...
                </h2>,
              ],
              "className": "col-xs-12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Random Quote Generator",
                  "className": "text-center",
                },
                "ref": null,
                "rendered": "Random Quote Generator",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Quotes are so inspiring...",
                  "className": "text-center",
                },
                "ref": null,
                "rendered": "Quotes are so inspiring...",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-xs-12"
            >
              <div
                className="text-center quote-box"
              >
                <div
                  className="quote"
                >
                  <h3
                    className="text-left"
                  >
                    <i
                      className="fa fa-quote-left fa-2x"
                    />
                  </h3>
                  <p
                    className="quote-content"
                  >
                    test
                  </p>
                  <h3
                    className="text-right"
                  >
                    <i
                      className="fa fa-quote-right fa-2x"
                    />
                  </h3>
                </div>
                <p
                  className="author"
                >
                  test
                </p>
              </div>
              <hr />
              <button
                className="btn btn-default"
                id="getQuote"
                onClick={[Function]}
              >
                Get Quote
              </button>
              <a
                className="btn btn-primary"
                href="https://twitter.com/intent/tweet?text=test"
                id="twitter"
                target="_blank"
              >
                <i
                  className="fa fa-twitter"
                />
                 Tweet
              </a>
            </div>,
            "className": "row quote-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="text-center quote-box"
                >
                  <div
                    className="quote"
                  >
                    <h3
                      className="text-left"
                    >
                      <i
                        className="fa fa-quote-left fa-2x"
                      />
                    </h3>
                    <p
                      className="quote-content"
                    >
                      test
                    </p>
                    <h3
                      className="text-right"
                    >
                      <i
                        className="fa fa-quote-right fa-2x"
                      />
                    </h3>
                  </div>
                  <p
                    className="author"
                  >
                    test
                  </p>
                </div>,
                <hr />,
                <button
                  className="btn btn-default"
                  id="getQuote"
                  onClick={[Function]}
                >
                  Get Quote
                </button>,
                <a
                  className="btn btn-primary"
                  href="https://twitter.com/intent/tweet?text=test"
                  id="twitter"
                  target="_blank"
                >
                  <i
                    className="fa fa-twitter"
                  />
                   Tweet
                </a>,
              ],
              "className": "col-xs-12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="quote"
                    >
                      <h3
                        className="text-left"
                      >
                        <i
                          className="fa fa-quote-left fa-2x"
                        />
                      </h3>
                      <p
                        className="quote-content"
                      >
                        test
                      </p>
                      <h3
                        className="text-right"
                      >
                        <i
                          className="fa fa-quote-right fa-2x"
                        />
                      </h3>
                    </div>,
                    <p
                      className="author"
                    >
                      test
                    </p>,
                  ],
                  "className": "text-center quote-box",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h3
                          className="text-left"
                        >
                          <i
                            className="fa fa-quote-left fa-2x"
                          />
                        </h3>,
                        <p
                          className="quote-content"
                        >
                          test
                        </p>,
                        <h3
                          className="text-right"
                        >
                          <i
                            className="fa fa-quote-right fa-2x"
                          />
                        </h3>,
                      ],
                      "className": "quote",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": <i
                            className="fa fa-quote-left fa-2x"
                          />,
                          "className": "text-left",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "host",
                          "props": Object {
                            "className": "fa fa-quote-left fa-2x",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "i",
                        },
                        "type": "h3",
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "test",
                          "className": "quote-content",
                        },
                        "ref": null,
                        "rendered": "test",
                        "type": "p",
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": <i
                            className="fa fa-quote-right fa-2x"
                          />,
                          "className": "text-right",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "host",
                          "props": Object {
                            "className": "fa fa-quote-right fa-2x",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "i",
                        },
                        "type": "h3",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "test",
                      "className": "author",
                    },
                    "ref": null,
                    "rendered": "test",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Get Quote",
                  "className": "btn btn-default",
                  "id": "getQuote",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Get Quote",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-twitter"
                    />,
                    " Tweet",
                  ],
                  "className": "btn btn-primary",
                  "href": "https://twitter.com/intent/tweet?text=test",
                  "id": "twitter",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-twitter",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " Tweet",
                ],
                "type": "a",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-12"
            >
              <h1
                className="text-center"
              >
                Random Quote Generator
              </h1>
              <h2
                className="text-center"
              >
                Quotes are so inspiring...
              </h2>
            </div>
          </div>
          <hr />
          <div
            className="row quote-container"
          >
            <div
              className="col-xs-12"
            >
              <div
                className="text-center quote-box"
              >
                <div
                  className="quote"
                >
                  <h3
                    className="text-left"
                  >
                    <i
                      className="fa fa-quote-left fa-2x"
                    />
                  </h3>
                  <p
                    className="quote-content"
                  >
                    test
                  </p>
                  <h3
                    className="text-right"
                  >
                    <i
                      className="fa fa-quote-right fa-2x"
                    />
                  </h3>
                </div>
                <p
                  className="author"
                >
                  test
                </p>
              </div>
              <hr />
              <button
                className="btn btn-default"
                id="getQuote"
                onClick={[Function]}
              >
                Get Quote
              </button>
              <a
                className="btn btn-primary"
                href="https://twitter.com/intent/tweet?text=test"
                id="twitter"
                target="_blank"
              >
                <i
                  className="fa fa-twitter"
                />
                 Tweet
              </a>
            </div>
          </div>
        </div>,
        "style": Object {
          "backgroundColor": "blue",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <h1
                  className="text-center"
                >
                  Random Quote Generator
                </h1>
                <h2
                  className="text-center"
                >
                  Quotes are so inspiring...
                </h2>
              </div>
            </div>,
            <hr />,
            <div
              className="row quote-container"
            >
              <div
                className="col-xs-12"
              >
                <div
                  className="text-center quote-box"
                >
                  <div
                    className="quote"
                  >
                    <h3
                      className="text-left"
                    >
                      <i
                        className="fa fa-quote-left fa-2x"
                      />
                    </h3>
                    <p
                      className="quote-content"
                    >
                      test
                    </p>
                    <h3
                      className="text-right"
                    >
                      <i
                        className="fa fa-quote-right fa-2x"
                      />
                    </h3>
                  </div>
                  <p
                    className="author"
                  >
                    test
                  </p>
                </div>
                <hr />
                <button
                  className="btn btn-default"
                  id="getQuote"
                  onClick={[Function]}
                >
                  Get Quote
                </button>
                <a
                  className="btn btn-primary"
                  href="https://twitter.com/intent/tweet?text=test"
                  id="twitter"
                  target="_blank"
                >
                  <i
                    className="fa fa-twitter"
                  />
                   Tweet
                </a>
              </div>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-xs-12"
              >
                <h1
                  className="text-center"
                >
                  Random Quote Generator
                </h1>
                <h2
                  className="text-center"
                >
                  Quotes are so inspiring...
                </h2>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="text-center"
                  >
                    Random Quote Generator
                  </h1>,
                  <h2
                    className="text-center"
                  >
                    Quotes are so inspiring...
                  </h2>,
                ],
                "className": "col-xs-12",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Random Quote Generator",
                    "className": "text-center",
                  },
                  "ref": null,
                  "rendered": "Random Quote Generator",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Quotes are so inspiring...",
                    "className": "text-center",
                  },
                  "ref": null,
                  "rendered": "Quotes are so inspiring...",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-xs-12"
              >
                <div
                  className="text-center quote-box"
                >
                  <div
                    className="quote"
                  >
                    <h3
                      className="text-left"
                    >
                      <i
                        className="fa fa-quote-left fa-2x"
                      />
                    </h3>
                    <p
                      className="quote-content"
                    >
                      test
                    </p>
                    <h3
                      className="text-right"
                    >
                      <i
                        className="fa fa-quote-right fa-2x"
                      />
                    </h3>
                  </div>
                  <p
                    className="author"
                  >
                    test
                  </p>
                </div>
                <hr />
                <button
                  className="btn btn-default"
                  id="getQuote"
                  onClick={[Function]}
                >
                  Get Quote
                </button>
                <a
                  className="btn btn-primary"
                  href="https://twitter.com/intent/tweet?text=test"
                  id="twitter"
                  target="_blank"
                >
                  <i
                    className="fa fa-twitter"
                  />
                   Tweet
                </a>
              </div>,
              "className": "row quote-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="text-center quote-box"
                  >
                    <div
                      className="quote"
                    >
                      <h3
                        className="text-left"
                      >
                        <i
                          className="fa fa-quote-left fa-2x"
                        />
                      </h3>
                      <p
                        className="quote-content"
                      >
                        test
                      </p>
                      <h3
                        className="text-right"
                      >
                        <i
                          className="fa fa-quote-right fa-2x"
                        />
                      </h3>
                    </div>
                    <p
                      className="author"
                    >
                      test
                    </p>
                  </div>,
                  <hr />,
                  <button
                    className="btn btn-default"
                    id="getQuote"
                    onClick={[Function]}
                  >
                    Get Quote
                  </button>,
                  <a
                    className="btn btn-primary"
                    href="https://twitter.com/intent/tweet?text=test"
                    id="twitter"
                    target="_blank"
                  >
                    <i
                      className="fa fa-twitter"
                    />
                     Tweet
                  </a>,
                ],
                "className": "col-xs-12",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="quote"
                      >
                        <h3
                          className="text-left"
                        >
                          <i
                            className="fa fa-quote-left fa-2x"
                          />
                        </h3>
                        <p
                          className="quote-content"
                        >
                          test
                        </p>
                        <h3
                          className="text-right"
                        >
                          <i
                            className="fa fa-quote-right fa-2x"
                          />
                        </h3>
                      </div>,
                      <p
                        className="author"
                      >
                        test
                      </p>,
                    ],
                    "className": "text-center quote-box",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <h3
                            className="text-left"
                          >
                            <i
                              className="fa fa-quote-left fa-2x"
                            />
                          </h3>,
                          <p
                            className="quote-content"
                          >
                            test
                          </p>,
                          <h3
                            className="text-right"
                          >
                            <i
                              className="fa fa-quote-right fa-2x"
                            />
                          </h3>,
                        ],
                        "className": "quote",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "host",
                          "props": Object {
                            "children": <i
                              className="fa fa-quote-left fa-2x"
                            />,
                            "className": "text-left",
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": null,
                            "key": null,
                            "nodeType": "host",
                            "props": Object {
                              "className": "fa fa-quote-left fa-2x",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "i",
                          },
                          "type": "h3",
                        },
                        Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "host",
                          "props": Object {
                            "children": "test",
                            "className": "quote-content",
                          },
                          "ref": null,
                          "rendered": "test",
                          "type": "p",
                        },
                        Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "host",
                          "props": Object {
                            "children": <i
                              className="fa fa-quote-right fa-2x"
                            />,
                            "className": "text-right",
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": null,
                            "key": null,
                            "nodeType": "host",
                            "props": Object {
                              "className": "fa fa-quote-right fa-2x",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "i",
                          },
                          "type": "h3",
                        },
                      ],
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "test",
                        "className": "author",
                      },
                      "ref": null,
                      "rendered": "test",
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get Quote",
                    "className": "btn btn-default",
                    "id": "getQuote",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Get Quote",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-twitter"
                      />,
                      " Tweet",
                    ],
                    "className": "btn btn-primary",
                    "href": "https://twitter.com/intent/tweet?text=test",
                    "id": "twitter",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-twitter",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " Tweet",
                  ],
                  "type": "a",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
